bin/add-extension-package.js

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env node
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: loopback-next
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+/**
+ * This is an internal script to add a package to `packages` or `extensions`
+ */
+'use strict';
+
+const build = require('../packages/build');
+const path = require('path');
+const cwd = process.cwd();
+
+let name = process.argv[2];
+if (name == null) {
+  console.error(
+    'Usage: %s <package-name>',
+    path.relative(cwd, process.argv[1]),
+  );
+  process.exit(1);
+}
+
+if (name.startsWith('@loopback/')) {
+  name = name.substring('@loopback/'.length);
+}
+
+const rootPath = path.resolve(__dirname, '..');
+let location = path.relative(rootPath, cwd);
+if (location !== 'packages' && location !== 'extensions') {
+  location = 'extensions';
+  process.chdir(path.join(rootPath, 'extensions'));
+}
+
+console.log('Adding project %s/%s...', location, name);
+
+// Run `lb4 extension`
+const child = build.runCLI(path.join(rootPath, 'packages/cli/bin/cli-main'), [
+  'extension',
+  `@loopback/${name}`,
+  '--outdir',
+  name,
+  '--yes',
+  '--skip-install',
+]);
+
```

Comment:
How about the remaining steps described in https://github.com/strongloop/loopback-next/blob/master/docs/site/DEVELOPING.md#adding-a-new-package?

Ideally, the script should run them all.

If that's too much work, then please print a descriptive message at the end to tell the user what are the next required steps in the package registration process, and include a link to our DEVELOPING docs.

---

Our docs are missing instructions for setting `repository.directory` field based on the location in the monorepo, see e.g. here:

https://github.com/strongloop/loopback-next/blob/1daa2b83bbd2474ab42d5227f4b9b97ef8b2e443/packages/core/package.json#L42-L46

Please make sure your tool is setting correctly all `repository` metadata.

Reply:
Fixed.